### PR TITLE
Support shallow cloning submodules

### PR DESCRIPTION
**Pull request by adborbas**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires MINOR [version update](https://semver.org/)

### Context

Support shallow cloning submodules to speed up cloning of repositories with submodules.

Resolves: [STEP-51](https://bitrise.atlassian.net/browse/STEP-51)

### Changes

- Updated `github.com/bitrise-io/go-utils/command/git`
- Introduced `limit_submodule_update_depth` input with default value of `yes`. If input has `yes` value that submodules will be cloned with the `--depth=1` argument.
- Create tests to cover the case.

### CODE REVIEW

**Review by istvankovacs-bitrise — DISMISSED** [review]

File: step.yml [hunk 1]
```diff
@@ -96,34 +96,42 @@ inputs:
...
     opts:
       category: Clone Config
       title: "Limit fetching to the specified number of commits"
       description: |-
         Limit fetching to the specified number of commits.
         The value should be a decimal number, for example `10`.
+  - limit_submodule_update_depth: "yes"
+    opts:
+      category: Clone Config
+      title: "Shalow update submodules"
```
Comment (lpusok) [inline step.yml:111]: Typos, please check with techwriters.

**Comment by lpusok** [comment]
Can you please link a sample workflow with a real repository, with this option applied?

**Comment by adborbas** [comment]
Good catch, I've forgotten to update the e2e tests. I will added the test case for this. 👍

**Comment by adborbas** [comment]
Actually, `_test_submodule` should already cover this case as both `update_submodules` and `limit_submodule_update_depth` has a default "yes" value. Do you think we are covered with that?

**Comment by lpusok** [comment]
@adborbas 
>  Do you think we are covered with that?

Based on https://stackoverflow.com/questions/2144406/how-to-make-shallow-git-submodules/17692710#17692710 shallow update could fail if the referenced commit is not a branch/tag but they added fixes to check out by commit hash (which not every git server supports). Can you check that the main providers work if the referenced submodule version is not on the tip of branch/tag? In https://github.com/bitrise-io/git-clone-test unsure which commit the submodule points to.

File: step.yml [hunk 2]
```diff
@@ -96,34 +96,42 @@ inputs:
...
       category: Clone Config
       title: "Limit fetching to the specified number of commits"
       description: |-
         Limit fetching to the specified number of commits.
         The value should be a decimal number, for example `10`.
+  - limit_submodule_update_depth: "yes"
+    opts:
+      category: Clone Config
+      title: "Shallow update submodules"
+      summary: "Shoud submodules be shallow cloned with a history truncated to the latest revisions."
```
Comment (lpusok) [inline step.yml:112]: typo in Shoud

**Review by lpusok — APPROVED** [review]
LGTM except one typo.

**Review by lpusok — APPROVED** [review]

**Pull request merged**